### PR TITLE
[11.0][ADD] purchase_representative

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
This module adds the field *Purchase Representative* to Purchase
Orders. This field is present in Standard Odoo v12. The user specified will
be the contact person for the purchase order.

Additionally, it defaults it to the user doing a procurement to overcome the
known flaw of Odoo (until v13) which changes the user in the transaction when
using `sudo`.

@ForgeFlow

### CODE REVIEW

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1:

File: purchase_representative/models/procurement_rule.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MrpInventory(models.Model):
```
Comment (etobella) [inline purchase_representative/models/procurement_rule.py]: MrpInventory?

File: purchase_representative/hooks.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tools import pycompat
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
```
Comment (etobella) [inline purchase_representative/hooks.py:10]: Maybe a post_uninstall_hook is necessary, no?

**Review by etobella — APPROVED** [review]
LGTM :+1: Just a few comments

File: purchase_representative/hooks.py (hunk 2 above)
Comment (LoisRForgeFlow) [reply purchase_representative/hooks.py:10]: Done!

File: purchase_representative/models/procurement_rule.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class ProcurementRule(models.Model):
+    _inherit = "procurement.rule"
+
+    @api.multi
+    def _run_buy(
+            self, product_id, product_qty, product_uom,
+            location_id, name, origin, values
+    ):
+        values["propagate_create_uid"] = self.env.uid
+        return super(MrpInventory, self)._run_buy(
```
Comment (etobella) [inline purchase_representative/models/procurement_rule.py]: Remove unnecessary please

File: purchase_representative/models/procurement_rule.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            product_id, product_qty, product_uom, location_id, name,
+            origin, values)
+
+    def _prepare_purchase_order(self, product_id, product_qty, product_uom,
+                                origin, values, partner):
+        vals = super(MrpInventory, self)._prepare_purchase_order(
```
Comment (etobella) [inline purchase_representative/models/procurement_rule.py]: Remove MrpInventory please

**Pull request merged**